scipy/stats/tests/test_sampling.py

```diff
@@ -319,6 +319,30 @@ def check_discr_samples(rng, pv, mv_ex):
     assert pval > 0.1
 
 
+def test_warning_center_not_in_domain():
+    # UNURAN will warn if the center provided or the one computed w/o the
+    # domain is outside of the domain
+    msg = "102 : center moved into domain of distribution"
+    with pytest.warns(RuntimeWarning, match=msg):
+        NumericalInversePolynomial(StandardNormal(), center=0, domain=(3, 5))
+    with pytest.warns(RuntimeWarning, match=msg):
+        NumericalInversePolynomial(StandardNormal(), domain=(3, 5))
```

Comment:
This test already would pass in `main`, but perhaps that's expected? Anyway, the rest wouldn't.

Reply:
Yes, I wanted to add a test case since the behaviour is slightly confusing, maybe one could even suppress the warning. In the end, setting the domain just triggers an update of center although the user did not even provide this parameter.